plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/GitHookTasksTest.kt

```diff
@@ -138,6 +138,30 @@ class GitHookTasksTest : AbstractPluginTest() {
         }
     }
 
+    @Test
+    internal fun `Format hook should not add non-indexed code to the commit`() {
+        // TODO: This test doesn't run git or verify that only indexed code is committed,
```

Comment:
Too bad JGit [does have](https://download.eclipse.org/jgit/site/5.11.0.202103091610-r/apidocs/index.html) partial staging :unamused:

Reply:
What do you mean?